source/service.go

```diff
@@ -214,6 +214,33 @@ func (sc *serviceSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, e
 		endpoints = append(endpoints, svcEndpoints...)
 	}
 
+	// this sorting is required to make merging work.
+	// after we merge endpoints that have same DNS, we want to ensure that we end up with the same service being an "owner"
+	// of all those records, as otherwise each time we update, we will end up with a different service that gets data merged in
+	// and that will cause external-dns to recreate dns record due to different service owner in TXT record.
```

Comment:
This is potentially not stable in case we introduce a new service, but I'm thinking that this might be fine in this case. WDYT? If you believe that we are okay with this, can you add a comment that covers this corner case?

Reply:
@Raffo agreed. I updated comment.